pkgs/tools/networking/socat/default.nix

```diff
@@ -8,6 +8,8 @@ stdenv.mkDerivation rec {
     sha256 = "0jnhjijyq74g3wa4ph0am83z6vq7qna7ac0xqjma8s4197z3zmhd";
   };
 
+  patches = stdenv.lib.optional stdenv.isDarwin ./speed-type-fix.patch;
```

Comment:
`stdenv.cc.isClang` would be a more correct, but I would personally conditionalize a patches like this one.

Reply:
In this case `isDarwin` is more accurate. The difference is in `termios.h`, in glibc it's `typedef unsigned int speed_t;`, so on Linux the types happen to be identical. On Darwin it's `typedef unsigned long speed_t;` which causes the error.